File: Python/5kyu/rot13.py
# Rot13
# https://www.codewars.com/kata/530e15517bc88ac656000716

# My code
def rot13(message):
    result = ""
    for v in message:
        c = ord(v)
        if v.islower(): c = c + 13 if c < ord('n') else c - 13
        elif v.isupper(): c = c + 13 if c < ord('N') else c - 13
        result += chr(c)
    return result

# Best code
# def rot13(message):
#     abc = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
#     cba = "nopqrstuvwxyzabcdefghijklmNOPQRSTUVWXYZABCDEFGHIJKLM"
#     mytable = message.maketrans(abc, cba)
#     return message.translate(mytable)

File: JavaScript/5kyu/rot13.js
// Rot13
// https://www.codewars.com/kata/530e15517bc88ac656000716

// My code
function rot13(message){
  let result = ""
  for (let i in message) {
    if (/[a-mA-M]/.test(message[i])) result += String.fromCharCode(message.charCodeAt(i)+13);
    if (/[n-zN-Z]/.test(message[i])) result += String.fromCharCode(message.charCodeAt(i)-13);
    if (!/[a-zA-Z]/.test(message[i])) result += message[i];
  }
  return result
}

// Best code
// function rot13(message){
//   return message.replace(/[a-zA-Z]/g, function(match) {
//     return String.fromCharCode(match.charCodeAt(0) + (match.toUpperCase().localeCompare('M') > 0 ? -13 : 13));
//   });
// }